sklearn/model_selection/_split.py

```diff
@@ -635,6 +635,122 @@ def split(self, X, y, labels=None):
+        Number of splits. Must be at least 1.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import HomogeneousTimeSeriesCV
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> htscv = HomogeneousTimeSeriesCV(n_splits=3)
+    >>> htscv.get_n_splits(X)
+    3
+    >>> print(htscv)  # doctest: +NORMALIZE_WHITESPACE
+    HomogeneousTimeSeriesCV(n_splits=3)
+    >>> for train_index, test_index in htscv.split(X):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [0] TEST: [1]
+    TRAIN: [0 1] TEST: [2]
+    TRAIN: [0 1 2] TEST: [3]
+
+    Notes
+    -----
+    The first fold has size ``n_samples // n_splits + n_samples % n_folds``,
+    other folds have size ``n_samples // n_splits``,
+    where ``n_samples`` is the number of samples.
+
+    Number of splitting iterations in this cross-validator, n_splits,
+    is not equal to other KFold based cross-validators'.
+    """
+    def __init__(self, n_splits=3):
+        super(HomogeneousTimeSeriesCV, self).__init__(n_splits,
+                                                      shuffle=False,
+                                                      random_state=None)
+
+    def split(self, X, y=None, labels=None):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            The target variable for supervised learning problems.
+
+        labels : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        Returns
+        -------
+        train : ndarray
+            The training set indices for that split.
+
+        test : ndarray
+            The testing set indices for that split.
+        """
+        X, y, labels = indexable(X, y, labels)
+        n_samples = _num_samples(X)
+        n_splits = self.n_splits
+        n_folds = n_splits + 1
+        if n_folds > n_samples:
+            raise ValueError(
+                ("Cannot have number of folds ={0} greater"
+                 " than the number of samples: {1}.").format(n_folds,
+                                                             n_samples))
+        indices = np.arange(n_samples)
+        fold_sizes = (n_samples // n_folds) * np.ones(n_folds, dtype=np.int)
+        fold_sizes[0] += n_samples % n_folds
+        cum_fold_sizes = np.cumsum(fold_sizes)
+        for start, stop in zip(cum_fold_sizes[:-1], cum_fold_sizes[1:]):
+            yield indices[:start], indices[start: stop]
+
+    def get_n_splits(self, X=None, y=None, labels=None):
```

Comment:
Yeah, it's not.

`get_n_splits` of `KFold` will return its number of folds.
`get_n_splits` of `HomogeneousTimeSeriesCV` will return number of folds - 1.